server/game/player.js

```diff
@@ -893,21 +894,11 @@ class Player extends Spectator {
     }
 
     moveCardToTopOfDeck(card, allowSave = true) {
-        this.game.applyGameAction('moveToTopOfDeck', card, card => {
-            this.game.raiseEvent('onCardReturnedToDeck', { player: this, card: card, allowSave: allowSave }, event => {
-                event.cardStateWhenMoved = card.createSnapshot();
-                this.moveCard(card, 'draw deck', { allowSave: allowSave });
-            });
-        });
+        return this.game.resolveGameAction(ReturnCardToDeck, { card, allowSave });
```

Comment:
The previous `'moveToTopOfDeck'` action is referenced in the implementation of When I Woke.

There are a couple options here:
1. Introduce game actions for both placing on top of deck and bottom of deck (could be same basic implementation, just need the game action name configurable)
2. Update When I Woke to reference "returnCardToDeck" instead
3. Remove the gameAction reference in When I Woke entirely.

I lean toward 3, since there are no cards that provide immunity from being placed on top of deck.

Reply:
I went with 3 as this was the easiest to do, also tested if when i woke still works